driver-core/src/main/com/mongodb/internal/graalvm/substitution/SocketStreamFactorySubstitution.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.internal.graalvm.substitution;
+
+import com.mongodb.ServerAddress;
+import com.mongodb.UnixServerAddress;
+import com.mongodb.internal.connection.SocketStreamFactory;
+import com.mongodb.internal.connection.Stream;
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import static com.mongodb.assertions.Assertions.fail;
+
+@TargetClass(SocketStreamFactory.class)
+public final class SocketStreamFactorySubstitution {
+    @Substitute
+    public Stream create(final ServerAddress serverAddress) {
+        if (serverAddress instanceof UnixServerAddress) {
+            throw new UnsupportedOperationException("UnixServerAddress is not supported in GraalVM native image");
```

Comment:
Assuming `UnixServerAddress` is not working properly in in GraalVM native image, this is not a breaking change. But this assumptions is based solely on the substitutions make in Quarkus. The same is true for `MongoCompressorSubstitution.createSnappyCompressor`.

Reply:
So, it might be possible to support it now. These substitutions are kind of old, like 4/5 years old and GraalVM evolved a lot in between. 

Note that the class will need to be marked as a runtime only class (cannot be initialized at build time)

Reply:
> Note that the class will need to be marked as a runtime only class (cannot be initialized at build time)

Could you please help me understand why? I made the snappy compressor work without any tricks, just by [collecting the reachability metadata with the tracing agent](https://www.graalvm.org/latest/reference-manual/native-image/metadata/AutomaticMetadataCollection/) via the [`org.graalvm.buildtools.native` Gradle plugin](https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html#agent-support). And the compression works regardless of whether I specify

- `--initialize-at-run-time=com.mongodb.internal.connection.SnappyCompressor`
- or `--initialize-at-build-time=com.mongodb.internal.connection.SnappyCompressor`